xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-docker/src/test/it/org/xwiki/flamingo/test/docker/AttachmentIT.java

```diff
@@ -448,19 +449,24 @@ void deleteAttachmentWithSpecialChar(TestUtils setup, TestReference testReferenc
 
     @Test
     @Order(9)
-    void checkEscapingInAttachmentName(TestUtils setup, TestReference testReference,
-        TestConfiguration testConfiguration)
+    void checkEscapingInAttachmentName(TestUtils setup, TestReference testReference) throws IOException
     {
         setup.loginAsSuperAdmin();
+
+        // We shouldn't store files with special characters in the repository, since some filesystems don't support it.
+        // Instead, we create the file during the test.
+        Path unescapedFile = Files.createTempFile("<strong>", null);
```

Comment:
Isn't this going to fail during the test (because it won't be able to create a file named like this) on Windows ?

Reply:
Not much we can do about that, since this test is specifically about testing this kind of filename no? At least the PR fixes the current situation preventing to perform the clone.

Reply:
> Not much we can do about that, since this test is specifically about testing this kind of filename no?

If the test is only about display, we can simply upload the attachment using something that let us choose the name of the attachment without the need to have an actual file on the filesystem (like REST).

Reply:
This test is about testing the actual JavaScript code for handling the upload. See [XWIKI-19611](https://jira.xwiki.org/browse/XWIKI-19611) for context.

Reply:
That's a good question. I would expect the `nio.Files` API to support such use cases and escape the name on NTFS filesystems, but I can't test it.
Having the name escaped would definitely make the test useless though. Do we support running Docker tests on Windows?

Reply:
> Do we support running Docker tests on Windows?

yes, in theory we should be able to build and test xwiki in any system

Reply:
> This test is about testing the actual JavaScript code for handling the upload.

OK, so that's indeed more complex. Maybe there would be some hack to have the browser end up with a different name than the actual file on filesystem...

Reply:
We didn't at the time since TC was not supporting it (it was very experimental), but I haven't retested it, maybe it works now: https://java.testcontainers.org/supported_docker_environment/windows/